packages/lit-html/src/directives/style-map.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {
+  AttributePart,
+  directive,
+  Directive,
+  noChange,
+  PartInfo,
+  ATTRIBUTE_PART,
+} from '../lit-html.js';
+
+/**
+ * A key-value set of CSS properties and values.
+ *
+ * The key should be either a valid CSS property name string, like
+ * `'background-color'`, or a valid JavaScript camel case property name
+ * for CSSStyleDeclaration like `backgroundColor`.
+ */
+export interface StyleInfo {
+  readonly [name: string]: string;
+}
+
+class StyleMap extends Directive {
+  previousStyleProperties?: Set<string>;
+
+  constructor(part: PartInfo) {
+    super();
+    if (
+      part.type !== ATTRIBUTE_PART ||
+      part.name !== 'style' ||
+      (part.strings !== undefined && part.strings.length > 2)
+    ) {
+      throw new Error(
+        'The `styleMap` directive must be used in the `style` attribute ' +
+          'and must be the only part in the attribute.'
+      );
+    }
+  }
+
+  render(styleInfo: StyleInfo) {
+    return Object.entries(styleInfo)
+      .map(([k, v]) => `${k}: ${v}`)
+      .join('; ');
+  }
+
+  update(part: AttributePart, [styleInfo]: Parameters<this['render']>) {
+    const {style} = part.element as HTMLElement;
+
+    if (this.previousStyleProperties === undefined) {
```

Comment:
It's unclear why we choose to actually use the string render for first render in `classMap` but not in `styleMap`. This is another case where `render` is _only_ implemented for SSR, and otherwise unused in the client code (and as such, why the tests didn't catch the camel-casing issue in `render`).

Reply:
I think we shouldn't block on addressing this as we've discussed modifying this.